apps/src/code-studio/projectRedux.js

```diff
@@ -47,6 +58,17 @@ export default (state = initialState, action) => {
       projectUpdatedStatus: action.status
     };
   }
+
+  if (action.type === SHOW_WORKSPACE_ALERT) {
+    return {
+      ...state,
+      showWorkspaceAlert: action.showWorkspaceAlert,
+      workspaceAlertType: action.workspaceAlertType,
+      workspaceAlertErrorMsg: action.workspaceAlertErrorMsg,
+      workspaceAlertDisplayBottom: action.workspaceAlertDisplayBottom
```

Comment:
this creates 4 separate pieces of state that are all related -- i think we should refactor this into 1 object that represents 1) that we have a workspace alert, and 2) its contents. something like

```javascript
return {
  ...state,
  workspaceAlert: {
    type: '',
    message: '',
    displayBottom: '',
  }
}
```

then we can use the presence of `state.project.workspaceAlert` to determine whether or not one is shown, and we use the contents of that object to determine what's in that workspace alert. and we can easily clean up this state when we no longer want an alert by setting `state.project.workspaceAlert` to null/undefined

Reply:
Great idea! This resolves my hesitation about adding type type and displayBottom to projectRedux file! Will do.